src/triton_object_recognition/src/object_recognizer.cpp

```diff
@@ -0,0 +1,313 @@
+#include <rcl_yaml_param_parser/parser.h>
+using std::placeholders::_1;
+using std::placeholders::_2;
+using std::placeholders::_3;
+using namespace std;
+using namespace cv;
+using namespace dnn;
+
+#define DEBUG_VISUALIZE 1
+
+namespace object_recognition
+{
+    ObjectRecognizer::ObjectRecognizer(const rclcpp::NodeOptions & options)
+    : Node("object_recognizer", options) 
+    {
+        //Create publisher, subscriber, and service
+        publisher_ = this->create_publisher<triton_interfaces::msg::DetectionBoxArray>("object_recognizer/out", 10);
+        subscription_ = image_transport::create_subscription(this,"object_recognizer/in",bind(&ObjectRecognizer::subscriberCallback, this, _1),"raw");
+        service_ = create_service<triton_interfaces::srv::ObjectDetection>("object_recognizer/recognize",bind(&ObjectRecognizer::serviceCallback, this, _1,_2));
+
+        //Populate parameters (values are not modified if parameters have not been declared)
+        this->declare_parameter("model_folder", model_folder_);
+        this->declare_parameter("weights_filename", weights_filename_);
+        this->declare_parameter("cfg_filename", cfg_filename_);
+        this->declare_parameter("weights_url", weights_url_);
+        this->declare_parameter("cfg_url", cfg_url_);
+        this->declare_parameter("classes", classes_);
+        this->declare_parameter("conf_threshold", conf_threshold_);
+        this->declare_parameter("nms_threshold", nms_threshold_);
+        this->declare_parameter("scale", scale_);
+        this->declare_parameter("inp_width", inp_width_);
+        this->declare_parameter("inp_height", inp_height_);
+        this->declare_parameter("swap_rb", swap_rb_);
+        this->declare_parameter("mean", mean_);
+        this->declare_parameter("backend", (int) backend_);
+        this->declare_parameter("target", (int) target_);
+
+        this->get_parameter("model_folder", model_folder_);
+        this->get_parameter("weights_filename", weights_filename_);
+        this->get_parameter("cfg_filename", cfg_filename_);
+        this->get_parameter("weights_url", weights_url_);
+        this->get_parameter("cfg_url", cfg_url_);
+        this->get_parameter("classes", classes_);
+        this->get_parameter("conf_threshold", conf_threshold_);
+        this->get_parameter("nms_threshold", nms_threshold_);
+        this->get_parameter("scale", scale_);
+        this->get_parameter("inp_width", inp_width_);
+        this->get_parameter("inp_height", inp_height_);
+        this->get_parameter("swap_rb", swap_rb_);
+        this->get_parameter("mean", mean_);
+        int backend, target;
+        if (this->get_parameter("backend", backend))
+            backend_ = (Backend) backend;
+        if (this->get_parameter("target", target))
+            target_ = (Target) target;
+
+        //Check model folder exists and creates if it doesn't
+        boost::filesystem::path model_folder = boost::filesystem::path(model_folder_);
```

Comment:
See comment in the config file, if we put a .gitignore in an empty triton_object_recognition/models directory, we can push that to git so everyone will have the models dir and we can use the ament_index function to retrieve from/add to it.